src/covid19sim/base.py

```diff
@@ -240,17 +243,25 @@ def _compute_preferences(self):
 
     def run(self, duration, outfile, start_time, all_possible_symptoms, port, n_jobs):
         self.current_day = 0
-
+        INTERVENTION_DAY = ExpConfig.get('INTERVENTION_DAY')
+        COLLECT_TRAINING_DATA = ExpConfig.get('COLLECT_TRAINING_DATA')
         print(f"INTERVENTION_DAY: {INTERVENTION_DAY}")
         while True:
             if self.env.timestamp.hour == 0:
                 # TODO: this is an assumption which will break in reality, instead of updating once per day everyone at the same time, it should be throughout the day
                 for human in self.humans:
                     human.uid = update_uid(human.uid, human.rng)
 
-            if INTERVENTION_DAY >= 0 and self.current_day == INTERVENTION_DAY:
+            if self.env.timestamp.hour == 0 and self.env.timestamp != self.env.initial_timestamp:
+                self.current_day += 1
```

Comment:
Why is it required to be here? I think day should increment at the end of the day... more likely after the `yield`

Reply:
I think Satya can answer better than I can, but it has to do with the change to multiple updates per day. Basically, people who update in early timeslots are going to have their risk calculated before they've updated their symptoms in the "at" loop.

Reply:
Actually, the run loop, here: https://github.com/pg2455/covid_p2p_simulation/pull/116/files#diff-b40966745ead579e546716d7d5c0ee0fR614

Reply:
This is related to the intervention code. The increment of the current day needs to be done before the intervention otherwise hour 0 is always associated to the previous day

Reply:
But this will not trigger intervention if `INTERVENTION_DAY=0`.